### ISSUE TITLE
[C#] Implement new Concept Exercise: constructors

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/v3#1226]
This issue describes how to implement the `constructors` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Constructors in C#.

## Learning objectives

- Know what constructors are.
- Know how to define parameterless constructors.
- Know how to define parameterized constructors.
- Know how to use constructor overloading.
- Know how to define private constructors.

## Out of scope

- Destructors.
- Static constructors.

## Concepts

The Concepts this exercise unlocks are:

- `constructors`: know what constructors are; know how to define parameterless constructors; know how to define parameterized constructors; know how to use constructor overloading; know how to define private constructors.

## Prequisites

This exercise's prerequisites Concepts are:

- `classes`: know how to work with classes.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Constructors][constructors]: introduction to constructors.
- [Constructor syntax][constructor-syntax]: syntax for defining constructors.

### After

- [Constructors][constructors]: introduction to constructors.
- [Constructor syntax][constructor-syntax]: syntax for defining constructors.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions][implementing-a-concept-exercise].

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[implementing-a-concept-exercise]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[constructors]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constructors
[constructor-syntax]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constructors#constructor-syntax
[using-constructors]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-constructors
[private-constructors]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/private-constructors

**Comment by ErikSchierboom** [comment]
Closed by #1389
